### PR TITLE
fix(core-flows,order): return damaged items

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
CLOSES: CC-363

### CODE REVIEW

File: integration-tests/http/__tests__/returns/returns.spec.ts [hunk 1]
```diff
@@ -973,6 +973,7 @@ medusaIntegrationTestRunner({
               items: [
                 expect.objectContaining({
                   received_quantity: 2,
+                  damaged_quantity: 1,
```
Comment (adrien2p) [inline integration-tests/http/__tests__/returns/returns.spec.ts:976]: suggestion: it sounds like this test is checking a lot of things and could grow more. should it be breaking down for more clarity about the different return?

**Review by adrien2p — APPROVED** [review]

File: integration-tests/http/__tests__/returns/returns.spec.ts (hunk 1 above)
Comment (fPolic) [inline integration-tests/http/__tests__/returns/returns.spec.ts:976]: question: so in this case, the user receives 3 items: 2 regular and one damaged? If so should we adjust the condition in confirm receive return to count damaged items when setting return status to partialy/fully received?

Comment (riqwan) [reply integration-tests/http/__tests__/returns/returns.spec.ts:976]: its testing a flow end to end, would be good to keep this as one test as each and every step before this affects the changes here and future changes

Comment (riqwan) [reply integration-tests/http/__tests__/returns/returns.spec.ts:976]: q: should we also test for inventory changes to ensure this is reflected correctly there?

**Review by riqwan — APPROVED** [review]
lgtm!

File: integration-tests/http/__tests__/returns/returns.spec.ts (hunk 1 above)
Comment (carlos-r-l-rodrigues) [reply integration-tests/http/__tests__/returns/returns.spec.ts:976]: The `received_quantity` will be always the total received, the regular ones and the damaged ones.
If 3 items are returned, and 1 is damaged, the object will contain:
```
{
  received_quantity: 3,
  damaged_quantity: 1,
}
```

Comment (adrien2p) [reply integration-tests/http/__tests__/returns/returns.spec.ts:976]: make sense, so you can always base on received quantity to not go over it for example 👍

Comment (carlos-r-l-rodrigues) [reply integration-tests/http/__tests__/returns/returns.spec.ts:976]: This is already been checked right below this.
That "damaged_quantity" counter is only to display things correctly in the UI, but the rules and inventory management haven't changed, they are fine.

**Pull request merged**